src/java.rs

```diff
@@ -0,0 +1,186 @@
+    if return_type.is_empty() {
+        return None;
+    }
+
+    let mut types: Vec<&str> = Vec::new();
+    let mut first_idx = 0;
+
+    let mut param_chrs = parameter_types.char_indices();
+    while let Some((idx, token)) = param_chrs.next() {
+        if token == 'L' {
+            let mut last_idx = idx;
+            for (i, c) in param_chrs.by_ref() {
+                last_idx = i;
+                if c == ';' {
+                    break;
+                }
+            }
+            let ty = parameter_types.get(first_idx..last_idx + 1)?;
+            if ty.is_empty() || !ty.ends_with(&[';']) {
+                return None;
+            }
+            types.push(ty);
+            first_idx = last_idx + 1;
+        } else if token == '[' {
+            continue;
+        } else if java_base_types(token).is_some() {
+            let ty = parameter_types.get(first_idx..idx + 1)?;
+            types.push(ty);
+            first_idx = idx + 1;
+        }
+    }
+
+    Some((types, return_type.to_string()))
+}
+
+/// returns a tuple where the first element is the list of the function
+/// parameters and the second one is the return type
+pub fn deobfuscate_bytecode_signature(
+    signature: &str,
+    mapper: &ProguardMapper,
+) -> Option<(Vec<String>, String)> {
+    let (parameter_types, return_type) = parse_obfuscated_bytecode_signature(signature)?;
+    let parameter_java_types: Vec<String> = parameter_types
+        .into_iter()
+        .filter(|params| !params.is_empty())
+        .filter_map(|params| byte_code_type_to_java_type(params, mapper))
+        .collect();
+
+    let return_java_type = byte_code_type_to_java_type(return_type.as_str(), mapper)?;
+
+    Some((parameter_java_types, return_java_type))
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::{java::byte_code_type_to_java_type, ProguardMapper, ProguardMapping};
+    use std::collections::HashMap;
+
+    #[test]
+    fn test_byte_code_type_to_java_type() {
+        let proguard_source = b"org.slf4j.helpers.Util$ClassContextSecurityManager -> org.a.b.g$a:
+    65:65:void <init>() -> <init>";
+
+        let mapping = ProguardMapping::new(proguard_source);
+        let mapper = ProguardMapper::new(mapping);
+
+        let tests = HashMap::from([
+            // invalid types
+            ("", ""),
+            ("L", ""),
+            ("", ""),
+            // valid types
+            ("[I", "int[]"),
+            ("I", "int"),
+            ("[Ljava/lang/String;", "java.lang.String[]"),
+            ("[[J", "long[][]"),
+            ("[B", "byte[]"),
+            (
+                // Obfuscated class type
+                "Lorg/a/b/g$a;",
+                "org.slf4j.helpers.Util$ClassContextSecurityManager",
+            ),
+        ]);
+
+        for (ty, expected) in tests {
+            assert_eq!(
+                byte_code_type_to_java_type(ty, &mapper).unwrap_or_default(),
+                expected.to_string()
```

Comment:
Instead of `unwrap_or_default`, you should rather check that the invalid types return `None`, and all the rest returns `Some(...)`